haystack/pipelines/base.py

```diff
@@ -1366,15 +1366,45 @@ def eval_batch(
             event_properties={"pipeline.classname": self.__class__.__name__, "pipeline.config_hash": self.config_hash},
         )
 
+        predictions_batches = self.run_batch(
+            queries=[label.query for label in labels], labels=labels, documents=documents, params=params, debug=True
+        )
+
+        eval_result = self._generate_eval_result_from_batch_preds(
+            predictions_batches=predictions_batches,
+            params=params,
+            sas_model_name_or_path=sas_model_name_or_path,
+            sas_batch_size=sas_batch_size,
+            sas_use_gpu=sas_use_gpu,
+            add_isolated_node_eval=add_isolated_node_eval,
+            custom_document_id_field=custom_document_id_field,
+            context_matching_min_length=context_matching_min_length,
+            context_matching_boost_split_overlaps=context_matching_boost_split_overlaps,
+            context_matching_threshold=context_matching_threshold,
+            use_auth_token=use_auth_token,
+        )
+
+        return eval_result
+
+    def _generate_eval_result_from_batch_preds(
+        self,
+        predictions_batches: Dict,
+        params: Optional[dict] = None,
+        sas_model_name_or_path: Optional[str] = None,
+        sas_batch_size: int = 32,
+        sas_use_gpu: bool = True,
+        add_isolated_node_eval: bool = False,
+        custom_document_id_field: Optional[str] = None,
+        context_matching_min_length: int = 100,
+        context_matching_boost_split_overlaps: bool = True,
+        context_matching_threshold: float = 65.0,
```

Comment:
Out of curiosity - where did these defaults like `context_matching_threshold` come from?